server/src/main/java/org/opensearch/index/IndexModule.java

```diff
@@ -641,4 +652,26 @@ private void ensureNotFrozen() {
         }
     }
 
+    public static Map<String, IndexStorePlugin.DirectoryFactory> createBuiltInDirectoryFactories(
+        Supplier<RepositoriesService> repositoriesService
+    ) {
+        final Map<String, IndexStorePlugin.DirectoryFactory> factories = new HashMap<>();
+        for (Type type : Type.values()) {
+            switch (type) {
+                case HYBRIDFS:
+                case NIOFS:
+                case FS:
+                case MMAPFS:
+                case SIMPLEFS:
+                    factories.put(type.getSettingsKey(), DEFAULT_DIRECTORY_FACTORY);
+                    break;
+                case REMOTE_SNAPSHOT:
```

Comment:
Should we add a feature flag check here before adding the `REMOTE_SNAPSHOT` properties?

Reply:
My only concern is it opens the code path up for use from a different location/code path. It will be an intentional config.

Sidenote: Is the standard convention to lock down just the API with feature flags?